### ISSUE TITLE
Свернуть все папки

### ISSUE THREAD

**Issue by tipodanet** [k-ivan/visual-bookmarks-chrome#182]
Иван, привет !
Известная проблема разработчиков -- не пользуются своими разработками. 
Как вам это удается ?  
Иван, сколько у вас папок в Закладках  ? 
10-20 ? 
У меня , наверное  , 1800 . Я смотрел файл Bookmarks , он там весь в JSON  . Как смог, я отделил закладки от их папок.
Не у всех папок 10-20 . Вы должны это учитывать .

**Comment by k-ivan** [comment]
> Известная проблема разработчиков -- не пользуются своими разработками.
Как вам это удается ?

Вы должны понимать, что все куйсы не предусмотришь и не оттестируешь, прежде чем утверждать это. Это не коммерческий продукт, которым я занимаюсь, как работой. Это фан проект.

Теперь по делу.

Чтобы заменить селект, нужно переписать его на кастомный компонент со своей логикой внутри. Я подумаю над этим, как будет побольше свободного времени. Это не 5-минутная задача.

**Comment by k-ivan** [comment]
И тут тоже не плохо было чуть больше мнений узнать, а надо ли оно.

**Comment by tipodanet** [comment]
> Чтобы заменить селект, нужно переписать его на кастомный компонент со своей логикой внутри.

да, я забыл -- Вы уже объясняли. Он стандартный. И папки все развернуты. Именно поэтому им не может пользоваться юзер, у кого больше 100 папок. Он ставит на самую популярную и больше не обращает на него внимание.

"Collapse All Folder"   -- тема настолько популярна, что встречается  везде: на форумах  браузеров, почтовых программ . Везде, где есть папки.  Если юзер пользует программу больше 10 лет, он обрастает сотнями папок. никуда от этого не деться.
Потом разрабы любезно применяют ИИ для помощи в авто-сортировке. :)  Полезно. И прежде, чем задействовать дорогой  ИИ , можно свернуть все папки, и запоминать их состояние. У каждого юзера сложится своя картина этого дерева. 
Это гораздо легче, чем ждать изобретения ИИ.

**Comment by k-ivan** [comment]
Нужно послушать больше мнений

Возможно уйти от решения селекта в пользу бокового меню из папок, там можно будет завезти туггл и при этом не потерять в UX.

**Comment by tipodanet** [comment]
В том недоделанном браузере Боковое Меню с закладками изначально сохраняло Состояние, и все свернуто, как надо . 
но у них есть еще желтый флажок в адресной строке с теми же функциями. Открывается на 6 строк, All Folders Expanded,  окно не растягивается. 
Я могу дать ссылку на ту тему на их форуме, но вы ее удалите :) Скажу просто: тема создана  в 2018 , папки свернуты в 2024. Это вопросу "Мы всегда идем навстречу пользователям" -- это у них в каждом релизе.
Иван, вам о не надо объяснять насколько это сложно  : ))) -- там в интерфейсе  какой-то атрибут за это отвечает. Т.е. размер исправления буквы 3-4.

**Comment by k-ivan** [comment]
> Возможно уйти от решения селекта в пользу бокового меню из папок, там можно будет завезти туггл и при этом не потерять в UX.

Возможно я реализую описанное в комментарии выше, но не ближайшее время, это пока не в приоритете.
